indico/modules/events/surveys/util.py

```diff
@@ -38,3 +38,17 @@ def make_survey_form(questions):
             continue
         setattr(form_class, name, field_impl.get_wtforms_field())
     return form_class
+
+
+def create_preview_form(questions):
+    """Creates a WTForm from survey questions
+
+    This form is intended to be used as a preview form of a survey
+
+    :return An `IndicoForm` subclass
+    """
+    form_class = type(b'SurveySubmissionForm', (IndicoForm,), {})
+    for question in questions:
+        setattr(form_class, question.title, question.field.wtf_field)
```

Comment:
Actually... wht do we need the separate function at all? I don't see any reason why `make_survey_form` can't be used for the preview, too (of course you need to adapt it to your changes - iirc you replaced `get_wtforms_field()` with a property)